source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -310,7 +233,220 @@ void TextureBaker::optimizeBakedTextures(NodePtr shader)
     }
 }
 
-DocumentPtr TextureBaker::bakeMaterial(NodePtr shader, const StringVec& udimSet)
+DocumentPtr TextureBaker::bakeMaterialToDoc(DocumentPtr doc, const FileSearchPath& searchPath, const string& materialPath, 
+                                              const StringVec udimSet, string& documentName)
```

Comment:
It doesn't look like the `documentName` argument is used by the method?

Reply:
I set the value for `documentName` on line 317. The original code would return a `std::pair` of the (`documentName`, `bakedDoc`). I thought it would be more intuitive to return `bakedDoc` and to pass `documentName` by reference and modify the value within the code.

Reply:
Ah, that makes sense.  Ok, let me take a second look at the interface of this method today, taking all of the latest changes into account.